tree.go

```diff
@@ -552,24 +570,105 @@ func (n *InternalNode) Commitment() *Point {
 }
 
 func (n *InternalNode) Commit() *Point {
-	poly := make([]Fr, NodeWidth)
-	emptyChildren := 256
-
+	if n.depth <= 1 {
```

Comment:
For `Commit()` of an `InternalNode` we do something similar. If we're at the root, or the next layer (i.e: depth <=1), we balance this work in all the cores available. 

We do this by calling a new `n.commitRoot()` and returning the result. What `commitRoot()` is just the parallelized version of the same work if `n.depth > 1` (lines below).

This is still a draft PR and we could make things a bit nicer maybe since there's some code repetition between the parallel version and the serial version. Also `CommitRoot()` isn't an ideal name since it's called for the top 2 layers, but you get the idea.